### DISCUSSION TITLE
Description of the predefined glsp-client elements?

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#538] [Q&A] [answered]
Where can I find a list or descriptions of the different elements predefined in [eclipse-glsp/glsp-client](https://github.com/eclipse-glsp/glsp-client)?

In the workflow-example the TaskNode extends the `RectangularNode` and implements `Nameable` and `WithEditableLabel`

What does this mean in detail? How can I figure out how the many different predefined glsp elements look like or behave? The main problem is that none of the source code files provides any comments. Is there somewhere a readme file? Reverse engeneering from the workflow-example is a very time consuming and error-prone way.

**Comment by planger** [comment, marked as answer]
Most of the [node types](https://github.com/eclipse/sprotty/blob/master/packages/sprotty/src/lib/model.ts) and [views](https://github.com/eclipse/sprotty/blob/master/packages/sprotty/src/lib/svg-views.tsx
) are located in Sprotty.

The same is true for the `Nameable` and `WithEditableLable` (see e.g. https://github.com/eclipse/sprotty/blob/master/packages/sprotty/src/features/edit/model.ts). These are extensions of the model with additional properties to enable certain additional client-side capabilities or behaviors, such as that you can invoke editing on a label or element that contains a label or that an element has a name and thus can be found by name.

Those features are typically structured as follows:
Each feature is defined by the files within a folder under `features`. In this folder, `model.ts` defines the model extensions required for the feature and the `di.config.ts` in the same folder registers the handlers, listeners, etc. that do something with this extension.
